### ISSUE TITLE
How can I control when the scanner is active?

### ISSUE THREAD

**Issue by burekas7** [juliansteenbakker/mobile_scanner#1004]
Hi,

I want to have control over when the scanner will start scanning and when it will stop.
How can this be done?

Currently the scanner starts scanning immediately when the screen comes up.

**Comment by burekas7** [comment]
It seems that the controller has that: **controller.start(), controller.stop()**
I will check it.

**Comment by burekas7** [comment]
@navaronbracke 
Thanks.
But if I get it right, the way to do that is what I mentioned, right? (start/stop)

**Comment by navaronbracke** [comment]
Yes, that is the way to use the API.

**Comment by burekas7** [comment]
@navaronbracke 

But when I call the `stop()` it turns off the camera.

My purpose is only to stop the scanning action, not the camera.

**Comment by navaronbracke** [comment]
In that case, we should handle pausing and resuming the `StreamSubscription` in `mobile_scanner`.

**Comment by burekas7** [comment]
@navaronbracke 
Thanks.

I did this local solution:

```
  void startListenToScannerCapture() {
    streamSubscription = widget.mobileScannerController.barcodes.listen(scannerlistener);
  }

  void scannerlistener(BarcodeCapture barcodeCapture) async {
    Some actions
    .
    .
    .
    .
    await streamSubscription.cancel();
  }

  // And when button pressed
        onPressed: () {
          startListenToScannerCapture();
        },

```

**Comment by burekas7** [comment]
@navaronbracke 

There is something weird here,
When I do: `streamSubscription.pause()`
And I move the scanner on a barcode, it doesn't read it, which is ok.
But when I `streamSubscription.resume()`, the stream listener callback is run with a scan value of the barcode while I was on pause.
Is it a bug?

**Comment by burekas7** [comment]
@navaronbracke 
Ok thanks, 
I will open a new issue for that.

**Comment by burekas7** [comment]
@navaronbracke 
Ok, so I closed #1008 and also added it as a comment in #1006
